agent/peering_endpoint_test.go

```diff
@@ -257,6 +292,76 @@ func TestHTTP_Peering_Read(t *testing.T) {
 	})
 }
 
+func TestHTTP_Peering_Delete(t *testing.T) {
+	if testing.Short() {
+		t.Skip("too slow for testing.Short")
+	}
+
+	t.Parallel()
+	a := NewTestAgent(t, "")
+
+	testrpc.WaitForTestAgent(t, a.RPC, "dc1")
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	// Insert peerings directly to state store.
+	// Note that the state store holds reference to the underlying
+	// variables; do not modify them after writing.
+	foo := &pbpeering.PeeringWriteRequest{
+		Peering: &pbpeering.Peering{
+			Name:                "foo",
+			State:               pbpeering.PeeringState_INITIAL,
+			PeerCAPems:          nil,
+			PeerServerName:      "fooservername",
+			PeerServerAddresses: []string{"addr1"},
+		},
+	}
+	_, err := a.rpcClientPeering.PeeringWrite(ctx, foo)
+	require.NoError(t, err)
+
+	t.Run("read existing token before attempting delete", func(t *testing.T) {
+		req, err := http.NewRequest("GET", "/v1/peering/foo", nil)
+		require.NoError(t, err)
+		resp := httptest.NewRecorder()
+		a.srv.h.ServeHTTP(resp, req)
+		require.Equal(t, http.StatusOK, resp.Code)
+
+		// TODO(peering): replace with API types
+		var pbresp pbpeering.Peering
+		require.NoError(t, json.NewDecoder(resp.Body).Decode(&pbresp))
+
+		require.Equal(t, foo.Peering.Name, pbresp.Name)
+	})
+
+	t.Run("delete the existing token we just read", func(t *testing.T) {
+		req, err := http.NewRequest("DELETE", "/v1/peering/foo", nil)
+		require.NoError(t, err)
+		resp := httptest.NewRecorder()
+		a.srv.h.ServeHTTP(resp, req)
+		require.Equal(t, http.StatusOK, resp.Code)
+		require.Equal(t, "{}", resp.Body.String())
+	})
+
+	t.Run("now the token is deleted, a read should 404", func(t *testing.T) {
+		req, err := http.NewRequest("GET", "/v1/peering/foo", nil)
+		require.NoError(t, err)
+		resp := httptest.NewRecorder()
+		a.srv.h.ServeHTTP(resp, req)
+		require.Equal(t, http.StatusNotFound, resp.Code)
+	})
+
+	t.Run("delete a token that does not exist", func(t *testing.T) {
+		req, err := http.NewRequest("DELETE", "/v1/peering/baz", nil)
+		require.NoError(t, err)
+		resp := httptest.NewRecorder()
+		a.srv.h.ServeHTTP(resp, req)
+
+		// TODO(peering): it may be a security concern, but do we want to say 404 here?
```

Comment:
No it shouldn't return 404 here, we return 200 for non-existent deletions in our APIs